### ISSUE TITLE
When xlwings opens new document with Bloomberg API formulas they are not working

### ISSUE THREAD

**Issue by pawlojr** [ZoomerAnalytics/xlwings#608]
I think when we open new document from python by xlwings somehow it blocks other addins like BBG one. In all cells where i had BDP, BDH, BDS in the opened from python file i see NAME errors. In general seems xlwings and BBG addin doens't like each other as in documents with a lot of BBG live formulas i get crashes of entire Excel from time to time

**Comment by fzumstein** [comment]
Please check if this resolves your issue: http://stackoverflow.com/questions/33125610/xlwings-wipes-out-my-excel-addin

**Comment by pawlojr** [comment]
Hi, Thanks. Unfortunately it didn't worked out, but I've found other solutions. I've tried to use the technique from that post to register all files that BBG addin is using to be sure that all is on - but with no effect. Anyway when i opened file from xlwings, to see if all of the addin components will be turn on (using File/Options/Add-ins/) both COMs and Excel addins looks like turned on. Yet when opening VBA editor i always seen that there are BloomgergUI and BloomgergUICharts files turned on as well, but if file was opened from xlwings it was nothing other than main file (even in the add-ins BloombergUI addin was seen as turned on). So what i did is before opening main file to just open C:\blp\API\Office Tools\BloombergUI.xla . Only then all was fine. Somehow if file is turned on from xlwings even BloombergUI.xla is turned on in the add-ins it's really not turn on at all and just by explicitly opening it fix the issue

**Comment by fzumstein** [comment]
Ok cool, I ll close the issue then since you could make it work.

**Comment by UlamNajk** [comment]
pawlojr, did you ever have an issue where Bloomberg was removed from the top ribbon after manipulating files through xlwings?

**Comment by pawlojr** [comment]
I have it all the time. Add-in is not loaded so all formulas are #Name, and nothing in the ribbon. Quick solutions is to just open add-in explicitly before opening main file like:
path_to_the_addin = "C:\\blp\\API\\Office Tools\\BloombergUI.xla"
wb_addin = xw.Book(path_to_the_addin)

which opens excel with that add-in loaded, then i can open any file and all bloomberg stuffs works fine

**Comment by UlamNajk** [comment]
That sounds like exactly what's happening for me, task manager to end the excel processes seems to fix it too.  There's no other way around it that you've seen?

**Comment by pawlojr** [comment]
For sure it is not related to Python and xlwings as it happens on my end also in the files without no python. I would say it is related with large amount of fields, specially live ones. And issue happens only after you close excel (like during closing it crash their addin). Also seems with newer versions of excel it happens more often. I switch to 2010 from 2016 and i see difference that it happens less times. And in 2010 if you will not open seperete excels processes but you will use file/open you will not have issue until you close it. I spoke with their help and their solutions was to just simplify files that i'm using or to move all data downloading to VBA. So I left digging into that closely. When I will have time i will replace all interaction with BBG by VBA but the excel file has already 100K lines :). If you find some reason pls lmk

**Comment by UlamNajk** [comment]
Ok, thanks for the insight.

**Comment by pawlojr** [comment]
I think I solved it on my end. Was digging today a bit around that. Was trying to find where exacly it happens (so what causing issue that after closing file and opening new excel making BBG addin disappear). I found finally one macro, which after running was always doing that issue. After tests I figured it out that it was turning off Events but didn't put Event on again. To have better optimalisation i was always turn off calculation, events, and screen update. And in that one macro i didn't turn event again. If that was last macro ran and then file was closed it was making issue 100% of times. When i added just line to turn events on at the end it was 100% correct. Also need to be sure that in case of any bug it will turn event on again in case it was set off at the begining. Maybe there are some other reasons but for now it works, making sure to close file with events set to ON. Somhow if it is off, then bbg addin was crashing. I reported that to their team.

**Comment by UlamNajk** [comment]
Ok, so AskToUpdateLinks and DisplayAlerts would fall under events, right?  Those are the only API elements I touch.

**Comment by hepbc** [comment]
Hi guys: have been facing this issue with another add-in. Tried to open the add-in explicitly as suggested by @pawlojr 

`wb = xw.Book(r"D:\Ace Equity Nxt\ActiveXl\ACEEQ_XL_NXT64.xll")
`
Get the following error:
`com_error: (-2147352567, 'Exception occurred.', (0, 'Microsoft Excel', 'Open method of Workbooks class failed', 'xlmain11.chm', 0, -2146827284), None)
`
Any help would be appreciated. 

@fzumstein for your attention too: Other add-ins don't load when I open a workbook through xlwings.

Thanks!

**Comment by fzumstein** [comment]
To load xlls, you need to do it like this:

```
wb.app.api.RegisterXLL(r"C:\path\to\YourAddIn.xll")
```

No add-ins are opened automatically, you need to do that yourself in the code.

**Comment by fzumstein** [comment]
Next time please open a new issue, closed ones usually get no attention.
